### ISSUE TITLE
RangeError: Invalid string length

### ISSUE THREAD

**Issue by jorgecasar** [google/wireit#543]
I'm having this bug in version 0.7.2:

```
[path/to/package-30:build:ts] Internal error! Please file a bug at https://github.com/google/wireit/issues/new, mention this message, that you encountered it in wireit version 0.7.2, and give information about your package.json files.
    Unknown error thrown: RangeError: Invalid string length
RangeError: Invalid string length
```

And with version 0.7.3 I'm having this other message:

```
file:///[...]/node_modules/wireit/lib/fingerprint.js:115
            this._str = JSON.stringify(this._data);
                             ^

RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at get string [as string] (file:///[...]/node_modules/wireit/lib/fingerprint.js:115:30)
    at LocalCache._getCacheDir (file:///[...]/node_modules/wireit/lib/caching/local-cache.js:49:104)
    at LocalCache.get (file:///[...]/node_modules/wireit/lib/caching/local-cache.js:18:31)
    at file:///[...]/node_modules/wireit/lib/execution/standard.js:66:42
    at async StandardScriptExecution._acquireSystemLockIfNeeded (file:///[...]/node_modules/wireit/lib/execution/standard.js:142:28)
    at async StandardScriptExecution._execute (file:///[...]/node_modules/wireit/lib/execution/standard.js:45:20)

Node.js v18.9.1
```

**Comment by aomarks** [comment]
Thanks for the report! Can you provide the relevant config?

**Comment by jorgecasar** [comment]
I really don't know what is the relevant config because in the previous commit it works fine. I only change typescript files of this package and I included some dependencies in other package. I hope you can understand it, because I can't share the real names.

Here is my config with normalized names. From one commit to the other I only added  `../package-[7-9]:build` in the wireit config of `package-10`. The package-30 is untouched and is the one that generates the error.

package-30:

```
"wireit": {
    "build": {
      "dependencies": [
        "build:ts",
        "build:bundle"
      ]
    },
    "build:bundle": {
      "command": "esbuild",
      "dependencies": [
        "build:ts"
      ],
      "files": [
        "development/**/*.{js,js.map,d.ts,d.ts.map}",
        "esbuild.config.js"
      ],
      "output": [
        "index.js",
        "index.js.map"
      ],
      "packageLocks": [
        "yarn.lock"
      ]
    },
    "build:ts": {
      "command": "tsc --build --pretty",
      "clean": "if-file-deleted",
      "files": [
        "src/**/*.ts",
        "tsconfig.json"
      ],
      "output": [
        "development/**/*.{js,js.map,d.ts,d.ts.map}",
        "tsconfig.tsbuildinfo"
      ],
      "dependencies": [
        "../package-1:build",
        "../package-[2-21]:build",
        "../package-22:build",
      ]
    }
}
```

package-10:

```
"wireit": {
    "build": {
      "dependencies": [
        "build:ts",
        "build:bundle"
      ]
    },
    "build:bundle": {
      "command": "esbuild",
      "dependencies": [
        "build:ts"
      ],
      "files": [
        "development/**/*.{js,js.map,d.ts,d.ts.map}",
        "esbuild.config.js"
      ],
      "output": [
        "index.js",
        "index.js.map"
      ],
      "packageLocks": [
        "yarn.lock"
      ]
    },
    "build:ts": {
      "command": "tsc --build --pretty",
      "clean": "if-file-deleted",
      "files": [
        "src/**/*.ts",
        "tsconfig.json"
      ],
      "output": [
        "development/**/*.{js,js.map,d.ts,d.ts.map}",
        "tsconfig.tsbuildinfo"
      ],
      "dependencies": [
        "../package-7:build",
        "../package-8:build",
        "../package-9:build",
        "../package-25:build",
        "../package-15:build",
        "../package-22:build"
      ]
    }
 }
 ```

**Comment by jorgecasar** [comment]
I tried adding new dependencies of package-10 one by one. Once I added the second I reach heap limit (JavaScript heap out of memory):

```
<--- Last few GCs --->

[71187:0x7f982504d000]    34089 ms: Scavenge 4004.4 (4117.5) -> 4004.3 (4118.5) MB, 7.3 / 0.0 ms  (average mu = 0.919, current mu = 0.666) allocation failure; 
[71187:0x7f982504d000]    34107 ms: Scavenge 4005.2 (4118.5) -> 4003.7 (4133.3) MB, 16.7 / 0.0 ms  (average mu = 0.919, current mu = 0.666) allocation failure; 
[71187:0x7f982504d000]    34331 ms: Mark-sweep 4018.4 (4133.3) -> 4016.6 (4149.3) MB, 211.5 / 0.0 ms  (average mu = 0.803, current mu = 0.316) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0x10af11de5 node::Abort() (.cold.1) [~/.nvm/versions/node/v18.9.1/bin/node]
 2: 0x1099dc059 node::Abort() [~/.nvm/versions/node/v18.9.1/bin/node]
 3: 0x1099dc23e node::OOMErrorHandler(char const*, bool) [~/.nvm/versions/node/v18.9.1/bin/node]
 4: 0x109b564e3 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [~/.nvm/versions/node/v18.9.1/bin/node]
 5: 0x109d1ef25 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [~/.nvm/versions/node/v18.9.1/bin/node]
 6: 0x109d1d902 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [~/.nvm/versions/node/v18.9.1/bin/node]
 7: 0x109d0fc3a v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [~/.nvm/versions/node/v18.9.1/bin/node]
 8: 0x109d105b5 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [~/.nvm/versions/node/v18.9.1/bin/node]
 9: 0x109cf21fa v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [~/.nvm/versions/node/v18.9.1/bin/node]
10: 0x109cebc19 v8::internal::MaybeHandle<v8::internal::SeqTwoByteString> v8::internal::FactoryBase<v8::internal::Factory>::NewRawStringWithMap<v8::internal::SeqTwoByteString>(int, v8::internal::Map, v8::internal::AllocationType) [~/.nvm/versions/node/v18.9.1/bin/node]
11: 0x10a17c49d v8::internal::IncrementalStringBuilder::Extend() [~/.nvm/versions/node/v18.9.1/bin/node]
12: 0x109e4b5d1 v8::internal::JsonStringifier::SerializeString(v8::internal::Handle<v8::internal::String>) [~/.nvm/versions/node/v18.9.1/bin/node]
13: 0x109e4f9a8 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
14: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
15: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
16: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
17: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
18: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
19: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
20: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
21: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
22: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
23: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
24: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
25: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
26: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
27: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
28: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
29: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
30: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
31: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
32: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
33: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
34: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
35: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
36: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
37: 0x109e50f74 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
38: 0x109e4e747 v8::internal::JsonStringifier::Result v8::internal::JsonStringifier::Serialize_<false>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
39: 0x109e482b3 v8::internal::JsonStringify(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>) [~/.nvm/versions/node/v18.9.1/bin/node]
40: 0x109be307c v8::internal::Builtin_JsonStringify(int, unsigned long*, v8::internal::Isolate*) [~/.nvm/versions/node/v18.9.1/bin/node]
41: 0x10a523239 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [~/.nvm/versions/node/v18.9.1/bin/node]
```

**Comment by aomarks** [comment]
Thank you! I think this must be a larger p
project than I've tested with before, so we're hitting up against limits in the size of JSON objects we can serialize.

Will work on a fix, which is likely to use hashes within the fingerprint file instead of embedding the full fingerprint of dependencies.

**Comment by aomarks** [comment]
Hi @jorgecasar, this should be fixed in `0.8.0` which I just released.

**Comment by jorgecasar** [comment]
Incredible speed solving issues. It works perfectly. Thank you so much!
